src/sdk/k4a.c

```diff
@@ -526,171 +526,252 @@ static k4a_result_t validate_configuration(k4a_context_t *device, const k4a_devi
+            LOG_ERROR("Neither depth camera nor color camera are enabled in the configuration, at least one needs to "
+                      "be enabled.",
+                      0);
         }
-
-        if (K4A_SUCCEEDED(result))
+        else
         {
-            result = K4A_RESULT_FROM_BOOL(config->synchronized_images_only == false);
+            if (config->depth_delay_off_color_usec != 0 || config->synchronized_images_only)
+            {
+                result = K4A_RESULT_FAILED;
+                LOG_ERROR("Both depth camera and color camera must be enabled to modify configuration of depth delay "
+                          "or synchronized image settings.",
+                          0);
+            }
         }
     }
 
-    if (K4A_SUCCEEDED(result) && depth_enabled)
+    if (K4A_SUCCEEDED(result))
     {
-        struct _depth_configuration
+        if (depth_enabled)
         {
-            k4a_depth_mode_t mode;
-            k4a_fps_t max_fps;
-        } supported_depth_configs[] = {
-            { K4A_DEPTH_MODE_NFOV_2X2BINNED, K4A_FRAMES_PER_SECOND_30 },
-            { K4A_DEPTH_MODE_NFOV_UNBINNED, K4A_FRAMES_PER_SECOND_30 },
-            { K4A_DEPTH_MODE_WFOV_2X2BINNED, K4A_FRAMES_PER_SECOND_30 },
-            { K4A_DEPTH_MODE_WFOV_UNBINNED, K4A_FRAMES_PER_SECOND_15 },
-            { K4A_DEPTH_MODE_PASSIVE_IR, K4A_FRAMES_PER_SECOND_30 },
-        };
-
-        bool depth_fps_and_mode_supported = false;
-        for (unsigned int x = 0; x < COUNTOF(supported_depth_configs); x++)
-        {
-            if (supported_depth_configs[x].mode == config->depth_mode &&
-                supported_depth_configs[x].max_fps >= config->camera_fps)
+            struct _depth_configuration
+            {
+                k4a_depth_mode_t mode;
+                k4a_fps_t max_fps;
+            } supported_depth_configs[] = {
+                { K4A_DEPTH_MODE_NFOV_2X2BINNED, K4A_FRAMES_PER_SECOND_30 },
+                { K4A_DEPTH_MODE_NFOV_UNBINNED, K4A_FRAMES_PER_SECOND_30 },
+                { K4A_DEPTH_MODE_WFOV_2X2BINNED, K4A_FRAMES_PER_SECOND_30 },
+                { K4A_DEPTH_MODE_WFOV_UNBINNED, K4A_FRAMES_PER_SECOND_15 },
+                { K4A_DEPTH_MODE_PASSIVE_IR, K4A_FRAMES_PER_SECOND_30 },
+            };
+
+            bool depth_fps_and_mode_supported = false;
+            for (unsigned int x = 0; x < COUNTOF(supported_depth_configs); x++)
             {
-                depth_fps_and_mode_supported = true;
-                break;
+                if (supported_depth_configs[x].mode == config->depth_mode &&
+                    supported_depth_configs[x].max_fps >= config->camera_fps)
+                {
+                    depth_fps_and_mode_supported = true;
+                    break;
+                }
+            }
+
+            if (!depth_fps_and_mode_supported)
+            {
+                result = K4A_RESULT_FAILED;
+                LOG_ERROR("The configured depth mode %d does not support fps mode %d.",
```

Comment:
This is tricky since the %d here will obscure the configuration. We should print the string values of the macros.